php/tests/generated_class_test.php

```diff
@@ -666,19 +723,80 @@ public function testPrefixMessage()
     # Test message with given namespace.
     #########################################################
 
-    public function testNamespaceMessage()
+    public function testNestedMessagesAndEnums()
     {
-        $m = new TestIncludeNamespaceMessage();
+        $m = new TestMessage();
+        $n = new TestMessage\Sub();
+        $m->setOptionalMessage($n);
+        $m->setOptionalNestedEnum(TestMessage\NestedEnum::ZERO);
+        $this->assertSame($n, $m->getOptionalMessage());
+        $this->assertSame(TestMessage\NestedEnum::ZERO, $m->getOptionalNestedEnum());
+    }
 
-        $n = new TestNamespace();
-        $n->setA(1);
-        $m->setNamespaceMessage($n);
-        $this->assertSame(1, $m->getNamespaceMessage()->getA());
+    public function testNestedMessagesAndEnumsWithPrefixes()
+    {
```

Comment:
Shall we merge this with testPrefixMessage and call it testNestedMessagesAndEnumsWithPrefix?